core/src/main/java/apoc/refactor/util/RefactorUtil.java

```diff
@@ -17,9 +19,13 @@ public static void mergeRelsWithSameTypeAndDirectionInMergeNodes(Node node, Refa
                     .filter(list -> !list.isEmpty())
                     .forEach(list -> {
                         Relationship first = list.get(0);
-                        for (int i = 1; i < list.size(); i++) {
-                            Relationship relationship = list.get(i);
-                            mergeRels(relationship, first, true,  config);
+                        if (isSelfRel(first) && !config.isSelfRel()) {
+                            list.forEach(Relationship::delete);
```

Comment:
scusa non ho capito una cosa perché se la prima è una relazione su se stessa poi le eliminiamo tutte? dobbiamo eliminare solo le self-rel

Reply:
prova a creare un test con il seguente dataset

```
(A), (B), (C)
(A)-[:KNOWS->(A)
(A)-[:KNOWS->(B)
(A)-[:KNOWS->(C)
(C)-[:KNOWS->(B)
```

in questo modo per come è implementatato qui cancella tutte le relazione di `(A)`

Reply:
@conker84 Sì ora ho fatto mente locale, prima di questo blocco c'è un `Collectors.groupingBy(rel -> Pair.of(rel.getStartNode(), rel.getEndNode())`, quindi le relazioni di uno specifico nodo vengono prima raggruppate per nodo di partenza ed arrivo, una volta che arrivo a quel `first` già so se tutti gli elementi della `list` sono o meno self-rel, e nel dataset che hai detto tu cancellerebbe solo la relazione `(A)-[:KNOWS->(A)`.
Effettivamente il nome `first` confonde, ma ho lasciato quello che già c'era